addons/sale_timesheet/controllers/portal.py

```diff
@@ -115,8 +115,9 @@ def _task_get_page_view_values(self, task, access_token, **kwargs):
         except (AccessError, MissingError):
             pass
 
-        if task.sale_order_id.invoice_ids:
-            moves = request.env['account.move'].search([('id', 'in', task.sale_order_id.invoice_ids.ids)])
+        moves = task.sale_order_id.invoice_ids
+        if moves and request.env['account.move'].check_access_rights('read', raise_exception=False):
+            moves = request.env['account.move'].search([('id', 'in', moves.ids)])
```

Comment:
moves will always be filled even if the user does not have access to it. This is not a security issue but we would redirect to an empty page. What do you think of the following?
```suggestion
        moves = self.env['account.move']
        invoice_ids = task.sale_order_id.invoice_ids
        if invoice_ids and request.env['account.move'].check_access_rights('read', raise_exception=False):
            moves = request.env['account.move'].search([('id', 'in', invoice_ids.ids)])
```

Reply:
@IT-Ideas , Done the changes. Thanks